### PR TITLE
niri: 0.1.3 -> 0.1.4

### PR DESCRIPTION
**Pull request by sodiboo**
## Description of changes

Updates niri to [`v0.1.4`](https://github.com/YaLTeR/niri/releases/tag/v0.1.4).

Additionally, removes the patch that was applied to 0.1.3; this is no longer necessary since it is already part of 0.1.4.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by sodiboo** [comment]
cc @IogaMaster: were we both making the same changes at the exact same time? (#300305)

**Comment by IogaMaster** [comment]
lol, I messed it up and made a second one.
We can move forward with your pr :smile:

**Comment by IogaMaster** [comment]
This review has been done with [warden](https://github.com/iogamaster/warden), please report any issues!


<details><summary>Packages built</summary>
<p>

✅ niri

### `nix-info -m` 
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.77, NixOS, 24.05 (Uakari), 24.05.20240216.5863c27`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Nix) 2.19.3`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`



</p>
</details>

**Review by IogaMaster — APPROVED** [review]
This review has been done with [warden](https://github.com/iogamaster/warden), please report any issues!


<details><summary>Packages built</summary>
<p>

✅ niri

### `nix-info -m` 
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.77, NixOS, 24.05 (Uakari), 24.05.20240216.5863c27`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Nix) 2.19.3`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`



</p>
</details>

**Comment by IogaMaster** [comment]
> Yeah, i could tell something was up with that first one. No worries, though. It's a neat coincidence that we were both submitting the same PR within minutes of one another, about 7 hours after the release was made.

I figured you weren't going to because of how long after the release. Guess i was wrong lol

File: pkgs/by-name/ni/niri/package.nix [hunk 1]
```diff
@@ -16,32 +16,25 @@
...
   };
 
-  patches = [
-    (fetchpatch {
-      name = "revert-viewporter.patch";
-      url = "https://github.com/YaLTeR/niri/commit/40cec34aa4a7f99ab12b30cba1a0ee83a706a413.patch";
-      hash = "sha256-3fg8v0eotfjUQY6EVFEPK5BBIBrr6vQpXbjDcsw2E8Q=";
-    })
-  ];
+  patches = [];
```
Comment (wegank) [inline pkgs/by-name/ni/niri/package.nix:32]:
```suggestion
```

**Review by IogaMaster — APPROVED** [review]

**Review by wegank — APPROVED** [review]

**Review by IogaMaster — APPROVED** [review]

**Pull request merged**